src/app/home-page/home-page.component.ts

```diff
@@ -11,7 +11,15 @@ export class HomePageComponent implements OnInit {
 
   public courses: CourseItemInfo[];
 
+  constructor(
+    private homePageService: HomePageService,
+  ) { }
+
   public ngOnInit(): void {
-    this.courses = fakeVideoList;
+    this.getCourses();
+  }
+
+  public getCourses(): void {
```

Comment:
consider implementing this functionality in a way  
public getCourses(): CourseItemInfo[] {
    return this.homePageService.getCourses();
  }
Is more straightforward and easier to read